src/NovelRT/Ecs/Input/InputSystem.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#include <NovelRT/Ecs/Ecs.h>
+
+namespace NovelRT::Ecs::Input
+{
+    InputSystem::InputSystem(std::shared_ptr<PluginManagement::IWindowingPluginProvider> windowingProvider,
+                             std::shared_ptr<PluginManagement::IInputPluginProvider> inputProvider)
+    {
+        _actionMap = std::map<NovelRT::Atom, std::string>();
+        _firstUpdate = true;
+        _device = inputProvider->GetInputService();
+        _device->Initialise(windowingProvider->GetWindowingDevice()->GetHandle());
+    }
+
+    void InputSystem::Update(Timing::Timestamp delta, Ecs::Catalogue catalogue)
+    {
+        if (_firstUpdate)
+        {
+            auto actions = _device->GetAllMappings();
+            for (auto&& act : actions)
```

Comment:
Please rename `act` to `action` if possible, or something that at least makes it a little clearer.